.github/workflows/publish.yml

```diff
@@ -57,28 +102,88 @@ jobs:
+        run: |
+          mkdir executables
+          mv ${{ env.PRODUCTION_TARGET }}/dmg/subspace-desktop_${{ steps.extract_version.outputs.version }}_x64.dmg executables/subspace-desktop_${{ steps.extract_version.outputs.version }}_${{ matrix.build.suffix }}.dmg
+          # Zip it so that signature is not lost
+          ditto -c -k --rsrc executables/subspace-desktop_${{ steps.extract_version.outputs.version }}_${{ matrix.build.suffix }}.dmg executables/subspace-desktop_${{ steps.extract_version.outputs.version }}_${{ matrix.build.suffix }}.zip
+          rm executables/subspace-desktop_${{ steps.extract_version.outputs.version }}_${{ matrix.build.suffix }}.dmg
+        if: matrix.build.target == 'x86_64-apple-darwin'
+
+      - name: Prepare installers for uploading (macOS arm)
+        run: |
+          mkdir executables
+          mv ${{ env.PRODUCTION_TARGET }}/dmg/subspace-desktop_${{ steps.extract_version.outputs.version }}_aarch64.dmg executables/subspace-desktop_${{ steps.extract_version.outputs.version }}_${{ matrix.build.suffix }}.dmg
+          # Zip it so that signature is not lost
+          ditto -c -k --rsrc executables/subspace-desktop_${{ steps.extract_version.outputs.version }}_${{ matrix.build.suffix }}.dmg executables/subspace-desktop_${{ steps.extract_version.outputs.version }}_${{ matrix.build.suffix }}.zip
+          rm executables/subspace-desktop_${{ steps.extract_version.outputs.version }}_${{ matrix.build.suffix }}.dmg
+        if: matrix.build.target == 'aarch64-apple-darwin'
+
+      - name: Prepare installers for uploading (Windows)
+        run: |
+          mkdir executables
+          move ${{ env.PRODUCTION_TARGET }}/msi/subspace-desktop_${{ steps.extract_version.outputs.version }}_x64_en-US.msi executables/subspace-desktop_${{ steps.extract_version.outputs.version }}_${{ matrix.build.suffix }}.msi
+          move ${{ env.PRODUCTION_TARGET }}/msi/subspace-desktop-opencl_${{ steps.extract_version.outputs.version }}_x64_en-US.msi executables/subspace-desktop-opencl_${{ steps.extract_version.outputs.version }}_${{ matrix.build.suffix }}.msi
+        if: runner.os == 'Windows'
+
+      - name: Upload executables to artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: executables-${{ matrix.build.suffix }}
+          path: |
+            executables/*
+          if-no-files-found: error
+
+      - name: Upload executables to assets
+        uses: alexellis/upload-assets@0.3.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
-          tagName: __VERSION__ # the action automatically replaces \_\_VERSION\_\_ with the app version
-          releaseName: "__VERSION__"
-          releaseBody: "See the assets to download this version and install."
-          releaseDraft: false
-          prerelease: true
+          asset_paths: '["executables/*"]'
+        if: github.event_name == 'push' && github.ref_type == 'tag'
```

Comment:
I believe either one or the other or both will fail unless you have proper `permissions` section in the workflow. Please check monorepo, it is there for a reason.

Reply:
In the monorepo, the `permissions` field is only there for the `container-linux` job (https://github.com/subspace/subspace/blob/main/.github/workflows/snapshot-build.yml#L22), not present for building the executables. Am I missing something here?

Reply:
Hm, either it affects other jobs anyway or it is indeed only necessary for container images and I got it mixed in my head. If it worked in test runs then ignore this comment.

Reply:
In the test runs, I did not create any release or pushed a tag, that's why this step is actually not tested. 
I don't think the the permissions set for one job will affect the other job, but that's only my guess :)
Is it ok to test this via creating a pre-release and then deleting that pre-release? If not, I will take a note of this, and in case of this permission error, we can quickly update the workflow

Reply:
I was testing such things in my personal fork. You can create as many test releases from whatever branch as you'd like. But let's no do such experiments in official repos if possible.

Reply:
Actually yes, I am quite certain it'll fail (you'll see familiar nicknames there): https://github.com/alexellis/upload-assets/issues/6

Probably GitHub uses the same token for multiple jobs in the same workflow or something, so it accidentally worked in monorepo.

Reply:
`permissions` were not required -> https://github.com/ozgunozerk/subspace-desktop/releases/tag/hoping
the errors they are getting the issue you linked, I have faced it as well. Maybe they did the same mistake as me (assumed that action would create the release).